src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -36,11 +38,19 @@ public class PushMetadataToBuildAssetRegistry : MSBuild.Task, ICancelableTask
 
         public string RepoRoot { get; set; }
 
+        public string AssetVersion { get; set; }
+
+        public string[] ManifestBuildData { get; set; }
+
+        public string PublishingVersion { get; set; } = "3";
```

Comment:
What are the implications of defaulting this to 3?

Reply:
Talked to @mmitche and he suggesting using 3 since this would be the "new" form of publishing since we'll include the manifest as a blob item compared to previous versions where this is not included. Other than that I don't think there are other implications

Reply:
The code generating the manifest in each build leg: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/src/GenerateBuildManifest.cs#L75

Is defaulting to the Legacy version right now. I think if only if the input versions were non-legacy would signing information end up in the manifest.  So perhaps keying off the input manifest versions would be the right thing to do. If those are all 3 (they shouldn't differ, and that should be checked here), then we would trigger the new merged manifest w/signing info behavior.

Does that make sense?

Basically what we're saying is the v1 is legacy (3.x), v2 is what we currently have, and v3 is the future. Any kind of breaking change (like this one), should go into 3. Using 3 requires that we have all v3 input manifests. Prior to this change, we would never had a merged manifest (the merging was implicit during publishing), so if all inputs were 3 then we're on 3. Now we're adding a new merged manifest (allowing us to ignore the other non-merged ones). It's version should match the publishing version of all input manifests.